tests/test-Image.php

```diff
@@ -0,0 +1,274 @@
+
+		$attachment_id = self::factory()->attachment->create_object(
+			$this->test_image, 0, [
+				'post_mime_type' => 'image/jpeg',
+				'post_type'      => 'attachment',
+			]
+		);
+
+		wp_update_attachment_metadata( $attachment_id, wp_generate_attachment_metadata( $attachment_id, $this->test_image ) );
+
+		$postmeta = get_post_meta( $attachment_id, '_wp_attachment_metadata', true );
+
+		$image = new A8C_Files\Image( $postmeta, $attachment_id );
+
+		$image->resize( $size );
+
+		// Resized image should be marked as resized.
+		$this->assertTrue( $image->is_resized() );
+		// Should have expected width.
+		$this->assertEquals( $expected_resize['width'], $image->get_width() );
+		// And height.
+		$this->assertEquals( $expected_resize['height'], $image->get_height() );
+		// Should have appropriate mime type.
+		$this->assertEquals( 'image/jpeg', $image->get_mime_type() );
+		// And should point to appropriate file.
+		$this->assertEquals( add_query_arg( $expected_resize['params'], 'image.jpg' ), $image->get_filename() );
+	}
+
+	/**
+	 * Unit test covering the get_filename method
+	 *
+	 * @covers A8C_Files\Image::get_filename
+	 */
+	public function test__get_filename() {
+		$attachment_id = self::factory()->attachment->create_object(
+			$this->test_image, 0, [
+				'post_mime_type' => 'image/jpeg',
+				'post_type'      => 'attachment',
+			]
+		);
+
+		wp_update_attachment_metadata( $attachment_id, wp_generate_attachment_metadata( $attachment_id, $this->test_image ) );
+
+		$postmeta = get_post_meta( $attachment_id, '_wp_attachment_metadata', true );
+
+		$image = new A8C_Files\Image( $postmeta, $attachment_id );
+
+		// Test getting original filename before the image is resized
+		$this->assertEquals( wp_basename( $this->test_image ), $image->get_filename() );
+
+		// Test resized filename
+		$image->resize( array( 'width' => 150, 'height' => 150, 'crop' => true ) );
+		$this->assertEquals( wp_basename( $this->test_image ) . '?resize=150,150', $image->get_filename() );
+	}
+
+	/**
+	 * Test get_resized_filename
+	 *
+	 * @covers A8C_Files\Image::get_resized_filename
+	 */
+	public function test__get_resized_filename() {
+		$attachment_id = self::factory()->attachment->create_object(
+			$this->test_image, 0, [
+				'post_mime_type' => 'image/jpeg',
+				'post_type'      => 'attachment',
+			]
+		);
+
+		wp_update_attachment_metadata( $attachment_id, wp_generate_attachment_metadata( $attachment_id, $this->test_image ) );
+
+		$postmeta = get_post_meta( $attachment_id, '_wp_attachment_metadata', true );
+
+		$image = new A8C_Files\Image( $postmeta, $attachment_id );
+
+		$width = $this->getProperty( 'width' );
+		$width->setValue( $image, 150 );
+
+		$height = $this->getProperty( 'height' );
+		$height->setValue( $image, 150 );
+
+		$get_resized_filename = $this->getMethod( 'get_resized_filename' );
+
+		$this->assertEquals( $this->test_image . '?resize=150,150', $get_resized_filename->invokeArgs( $image, [] ) );
```

Comment:
Minor: I find it's cleaner to keep logic and execution out of assertion calls. It helps with readability and debugging.

One testing pattern that helps with this (and is always nice to follow) is arrange-act-assert (e.g. http://wiki.c2.com/?ArrangeActAssert). It encourages a visual and structural separation of the various stages of a test:

```
// Arrange: setup the test condition
$attachment_id = self::factory()->attachment->create_object(
...
);
wp_update_attachment_metadata( $attachment_id, wp_generate_attachment_metadata( $attachment_id, $this->test_image ) );
...

// Act: the method under test
$get_resized_filename = $this->getMethod( 'get_resized_filename' );
$actual_resized_filename = $get_resized_filename->invokeArgs( $image, [] )

// Assert: did it work?
$this->assertEquals( $this->test_image . '?resize=150,150', $actual_resized_filename );
```

 A good tip I once heard was that you should be able to squint at a test method and visually be able to separate the three stages. It doesn't always work as cleanly as this (some tests are just too complex) but it's a good guideline to try and follow.